src/Hedgehog.Experimental/Gen.fs

```diff
@@ -597,8 +597,16 @@ module GenX =
               }
 
         | Shape.CliMutable (:? ShapeCliMutable<'a> as shape) ->
+            let getDepth (sm: IShapeMember<_>) =
+              let rec loop (t: Type) depth =
+                if t = null
+                then depth
+                else loop t.BaseType (depth + 1)
+              loop sm.MemberInfo.DeclaringType 0
             shape.Properties
-            |> Seq.toList
+            |> Array.toList
+            |> List.groupBy (fun p -> p.MemberInfo.Name)
+            |> List.map (fun (_, ps) -> ps |> List.sortByDescending getDepth |> List.head)
```

Comment:
I wonder if `snd >> List.sortByDescending getDepth >> head` can be replaced with `snd >> Seq.maxBy getDepth` for clarity.

Also for efficiency we can skip calling to `getDepth` if there is only one member in the group.

Sorry if it looks like nitpicking, but I see big projects with thousands of tests, and unnecessary work does add up to a significant time in running tests / CI...